pkg/registry/search/storage/cache.go

```diff
@@ -140,8 +140,8 @@ func addAnnotationWithClusterName(resourceObjects []runtime.Object, clusterName
 		if annotations == nil {
 			annotations = make(map[string]string)
 		}
-		// TODO: move this annotation key `cluster.karmada.io/name` to the Cluster API.
-		annotations["cluster.karmada.io/name"] = clusterName
+
+		annotations[clusterv1alpha1.ClusterAnnotationKeyName] = clusterName
```

Comment:
This annotation indicates which cluster a `cached resource` comes from.
So, I think the sub-domain should not be `cluster`, but `resource`.

My suggestion is:
```golang
	// CacheSourceAnnotationKey is the annotation that added to a resource to
	// represent which cluster it cached from.
	CacheSourceAnnotationKey = "resource.karmada.io/cached-from-cluster"
```

PS: `cluster` API might not be the most appropriate place to put this annotation, but since we don't have any other better place. So, it's a tradeoff here.

Reply:
ok, thanks point it.  `CacheSourceAnnotationKey = "resource.karmada.io/cached-from-cluster"` is reasonable.